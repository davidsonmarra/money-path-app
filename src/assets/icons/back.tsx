import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BackIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-back">
    <Path d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z" />
  </Svg>
);

export default BackIcon;
