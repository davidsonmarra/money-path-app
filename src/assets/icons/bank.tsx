import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BankIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-bank">
    <Path d="M200-280v-280h80v280zm240 0v-280h80v280zM80-120v-80h800v80zm600-160v-280h80v280zM80-640v-80l400-200 400 200v80zm178-80h444zm0 0h444L480-830z" />
  </Svg>
);

export default BankIcon;
