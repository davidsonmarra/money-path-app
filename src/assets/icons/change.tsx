import React from 'react';
import Svg, { Path } from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const ChangeIcon = ({ size = 24, color = '#FFF' }: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-change"
  >
    <Path d="m482-200 114-113-114-113-42 42 43 43q-28 1-54.5-9T381-381q-20-20-30.5-46T340-479q0-17 4.5-34t12.5-33l-44-44q-17 25-25 53t-8 57q0 38 15 75t44 66 65 43.5 74 15.5l-38 38zm165-170q17-25 25-53t8-57q0-38-14.5-75.5T622-622t-65.5-43-74.5-14l38-39-42-42-114 113 114 113 42-42-44-44q27 0 55 10.5t48 30.5 30.5 46 10.5 52q0 17-4.5 34T603-414zM480-80q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default ChangeIcon;
