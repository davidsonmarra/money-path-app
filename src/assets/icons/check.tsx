import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const CheckIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-check">
    <Path d="m424-296 282-282-56-56-226 226-114-114-56 56zm56 216q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default CheckIcon;
