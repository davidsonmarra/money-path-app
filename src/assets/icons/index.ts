export {default as AppleIcon} from 'src/assets/icons/apple';
export {default as BackIcon} from 'src/assets/icons/back';
export {default as BancoDoBrasilIcon} from 'src/assets/icons/banco-do-brasil';
export {default as BankIcon} from 'src/assets/icons/bank';
export {default as CheckIcon} from 'src/assets/icons/check';
export {default as ChevronRightIcon} from 'src/assets/icons/chevron-right';
export {default as CloseIcon} from 'src/assets/icons/close';
export {default as ErrorIcon} from 'src/assets/icons/error';
export {default as GoogleIcon} from 'src/assets/icons/google';
export {default as InterIcon} from 'src/assets/icons/inter';
export {default as ItauIcon} from 'src/assets/icons/itau';
export {default as MoneyPathIcon} from 'src/assets/icons/money-path';
export {default as NubankIcon} from 'src/assets/icons/nubank';
export {default as SantanderIcon} from 'src/assets/icons/santander';
export {default as WalletIcon} from 'src/assets/icons/wallet';
export {default as WarningIcon} from 'src/assets/icons/warning';
