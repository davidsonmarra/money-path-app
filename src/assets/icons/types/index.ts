export enum IconType {
  apple = 'apple',
  back = 'back',
  bancoDoBrasil = 'bancoDoBrasil',
  bank = 'bank',
  check = 'check',
  chevronRight = 'chevronRight',
  close = 'close',
  error = 'error',
  food = 'food',
  games = 'games',
  google = 'google',
  health = 'health',
  help = 'help',
  house = 'house',
  inter = 'inter',
  itau = 'itau',
  moneyPath = 'moneyPath',
  money = 'money',
  nubank = 'nubank',
  santander = 'santander',
  transport = 'transport',
  wallet = 'wallet',
  warning = 'warning',
}
