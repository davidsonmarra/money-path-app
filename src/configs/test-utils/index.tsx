import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
