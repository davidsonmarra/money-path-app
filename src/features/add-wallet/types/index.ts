import {IconType} from 'src/assets/icons/types';

export interface WalletProps {
  name: string;
  icon: IconType;
  color?: string;
  backgroundColor?: string;
  type: 'personal' | 'bank';
  amount?: number;
}

export type SelectWalletSection = {
  title: string;
  data: WalletProps[];
};
