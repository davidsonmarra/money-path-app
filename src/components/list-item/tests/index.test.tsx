import React from 'react';
import {fireEvent, render, screen} from 'src/configs/test-utils';
import ListItem, {Props as ListItemProps} from 'src/components/list-item';
import {IconType} from 'src/assets/icons/types';
import {WalletIcon} from 'src/assets/icons';
import renderIcon from 'src/assets/icons/utils';

const mockContent: ListItemProps['content'] = {
  type: 'simple',
  label: 'Label',
  description: 'Description',
};
const renderComponent = ({
  content = mockContent,
  onPress,
  ...rest
}: Partial<ListItemProps>) =>
  render(<ListItem content={content} onPress={onPress} {...rest} />);

describe('ListItem', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when clicked', () => {
    const mockOnPress = jest.fn();
    renderComponent({onPress: mockOnPress});
    fireEvent.press(screen.getByText(mockContent.label));
    expect(mockOnPress).toHaveBeenCalled();
  });

  describe('when content is simple', () => {
    it('renders label and description', () => {
      renderComponent({content: mockContent});
      expect(screen.getByText(mockContent.label)).toBeTruthy();
      expect(screen.getByText(mockContent.description!)).toBeTruthy();
    });

    it('renders third line if provided', () => {
      const thirdLine = 'Third line';
      renderComponent({content: {...mockContent, thirdLine}});
      expect(screen.getByText(thirdLine)).toBeTruthy();
    });
  });

  describe('when content is display', () => {
    const mockDisplayContent: ListItemProps['content'] = {
      type: 'display',
      label: 'Label',
      value: 'Value',
    };

    it('renders label and value', () => {
      renderComponent({content: mockDisplayContent});
      expect(screen.getByText(mockDisplayContent.label)).toBeTruthy();
      expect(screen.getByText(mockDisplayContent.value!)).toBeTruthy();
    });
  });

  describe('when leading is provided', () => {
    it('renders leading element', () => {
      renderComponent({
        leading: renderIcon(IconType.wallet)({color: 'red', size: 24}),
      });
      expect(screen.getByTestId('icon-wallet')).toBeTruthy();
    });
  });

  describe('when trailing is provided', () => {
    it('renders trailing element', () => {
      renderComponent({
        trailing: renderIcon(IconType.wallet)({color: 'red', size: 24}),
      });
      expect(screen.getByTestId('icon-wallet')).toBeTruthy();
    });
  });

  describe('when navigationIndicator is provided', () => {
    it('renders navigation indicator', () => {
      renderComponent({onPress: jest.fn(), navigationIndicator: true});
      expect(screen.getByTestId('icon-chevron-right')).toBeTruthy();
    });

    it('does not render navigation indicator if onPress is not provided', () => {
      renderComponent({navigationIndicator: true, onPress: undefined});
      expect(screen.queryByTestId('icon-chevron-right')).toBeNull();
    });
  });
});
