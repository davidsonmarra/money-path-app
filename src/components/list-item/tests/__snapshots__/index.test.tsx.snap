// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
    >
      <View>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 16,
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          Label
        </Text>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 16,
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          Description
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
